pkg/controller/storagecluster/cephcluster.go

```diff
@@ -343,8 +345,17 @@ func newExternalCephCluster(sc *ocsv1.StorageCluster, cephImage string, monitori
 	return externalCephCluster
 }
 
-func getMonCount(nodeCount int) int {
-	count := defaults.MonCountMin
+func getMinNodes(sc *ocsv1.StorageCluster) int {
+
+	if arbiterEnabled(sc) {
+		// We need a minimum of 2 OSDs in each of the two zones when arbiter is enabled
+		return defaults.ArbiterModeDeviceSetReplica * 2
```

Comment:
`defaults.ArbiterModeDeviceSetReplica` is already 2. Why do we need to multiply by 2?

Reply:
quoting from the definition:
```
ArbiterModeDeviceSetReplica is the default number of Rook-Ceph StorageClassDeviceSets per StorageCluster StorageDeviceSet when arbiter is enabled
```

In normal mode we have `3`here.
Now, we want 2 replicas per zone. (So that we have two copies in each zone.) 
But I think we want to make the constant factor of `2`here configurable too.
I think for this release it might be ok, but later, we want to be able to do one copy per zone...

Reply:
Good question:
There are two equations that we need
`minDevices  == minNodes == replicaPerFailureDomain * numFailureDomains`
`minAdditionalDevicesForExpansion == minAdditionalNodesPerExpansion == numFailureDomains`

1. In the non-arbiter cases, the minimum requirement is replica 1 within the failure domain and 3 failure domains in the cluster. To expand the cluster, you can start with the addition of 1 OSD per failure domain. Therefore,
`minDevices  == minNodes == 1 * 3`
`minAdditionalDevicesForExpansion == minAdditionalNodesPerExpansion == 3`

2. In the arbiter case, however, the minimum requirement is replica 2 within the failure domain and 2 failure domains in the cluster. To expand the cluster, you can start with the addition of 1 OSD per failure domain. Therefore,
`minDevices  == minNodes == 2 * 2`
`minAdditionalDevicesForExpansion == minAdditionalNodesPerExpansion == 2`

With that explanation, and to incorporate what @obnoxxx suggested, I will update the code to do the following:
1. add a default for ReplicaPerFailureDomain for arbiter and non arbiter configurations
2. add functions that capture the above equations

Reply:
Updated